lib/c-api/src/wasm_c_api/instance.rs

```diff
@@ -17,8 +17,7 @@ pub unsafe extern "C" fn wasm_instance_new(
     _store: &wasm_store_t,
     module: &wasm_module_t,
     imports: &wasm_extern_vec_t,
-    // own
```

Comment:
Might be useful to put this comment back if it's correct, otherwise it's not clear what the ownership is of `*mut *mut`, `own` means we're responsible for freeing it I believe

Reply:
We can't own it. We can't free it. The `start.c` example explicitely drops the `trap` value:

https://github.com/wasmerio/wasmer/blob/86f3438acd23ccb3026143bcebd96b49ddf659f3/lib/c-api/tests/wasm_c_api/wasm-c-api/example/start.c#L58-L60

and later:

https://github.com/wasmerio/wasmer/blob/86f3438acd23ccb3026143bcebd96b49ddf659f3/lib/c-api/tests/wasm_c_api/wasm-c-api/example/start.c#L95

I believe that the value is owned by the caller, not by the callee.

Reply:
Huh, `own wasm_trap_t**` is weird in the context of using arguments as outputs.

It seems like there's no need for the `Vec` at all then, this just seems to be a double pointer for the sake of setting a `trap*`

Reply:
So `wasm_trap_t**` must not be interpreted as an array of `wasm_trap_t*`, hmm. Let's fix that then.